03_it_analysis/src/it_analysis_data_prep.py

```diff
@@ -36,4 +39,112 @@ def download_s3_to_local(s3_dir_prefix, local_outdir, file_id):
         local_fpath = os.path.join(local_outdir,obj['Key'].split('/')[2])
         s3_client.download_file(s3_bucket, s3_fpath, local_fpath)
         print(s3_fpath+' Downloaded to local')
+
+#%%
+def select_sources(srcs, date_start, date_end):
+    '''select the variables you are interested in examining, 
+    srcs must be a list using the exact variable names,
+    it will return a list of dataframes, each dataframe corresponding to a site
+    with the requested variables as columns'''
+    
+    srcs_list = list()
+    
+    for file in os.listdir('02_munge/out/'):
+        #read each file
+        data = pd.read_csv('02_munge/out/'+file)
+        #print(file)
+        #print(data.head())
+        sources = list()
+        #if the columns of the dataframe contain any of the entries in srcs
+        if data.columns.to_series().str.contains(srcs[0]).any(): 
+            #select the columns that contain the entries in srcs
+            for s in srcs:
+                sources.append(data.loc[:,data.columns.to_series().str.contains(s)].columns[0])
+                #print(sources)
+           #subset those columns
+            data_col_select = data.loc[:,sources]
+            data_col_select = data_col_select.set_index(pd.to_datetime(data['datetime']))
+            data_col_select = data_col_select[date_start:date_end]
+            #this relies on a specific file naming structure, it appends the site name to the column header
+            data_col_select = data_col_select.add_suffix('_'+str(file.split('_')[2].split('.')[0]))
+            print(str(file.split('_')[2].split('.')[0])+' : Sources Found')
+            srcs_list.append(data_col_select)
+        else:
+            print(str(file.split('_')[2].split('.')[0])+' : No Data')
+            continue
+    return srcs_list
+    
+#%%
+def select_sinks(snks, date_start, date_end):
+    '''This is a filler function, for now it is hard coded and very specific to
+    the salt front location spreadsheet we have, but in the future it should look
+    like the select_sources function from above'''
+    
+    snks_list = list()
+
+    sf_loc = pd.read_csv('99_scratch/drb_estuary_salinity_scratch/data/saltfront.csv', index_col = 'datetime')
+    sf_loc.index = pd.to_datetime(sf_loc.index)
+    sf_loc = sf_loc[date_start:date_end]
+    #we'll make snks_list a list of df here they are both 2019, but these could be different model runs 
+    snks_list.append(sf_loc[snks].loc['2019'])
+    snks_list.append(sf_loc[snks].loc['2019'])
```

Comment:
should this be generalized to use the years from the function inputs `date_start` and `date_end`? Maybe it's not worth your time for now if you think this function will be removed or significantly changed later